internal/server/server.go

```diff
@@ -66,12 +70,24 @@ func NewServer(logger *logrus.Logger, client cloudapi.OsbuildClient, dbase db.DB
 	if err != nil {
 		panic(err)
 	}
+
+	loader := openapi3.NewSwaggerLoader()
+
+	if err := spec.Validate(loader.Context); err != nil {
+		panic(err)
+	}
+
+	router, err := legacyrouter.NewRouter(spec)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
nitpick, but I'd space it like this:

```
        ...

        loader := openapi3.NewSwaggerLoader()
	if err := spec.Validate(loader.Context); err != nil {
		panic(err)
	}
	router, err := legacyrouter.NewRouter(spec)
	if err != nil {
		panic(err)
	}

        majorVersion := strings.Split(spec.Info.Version, ".")[0]
```

just to have each paragraph do a single thing.